src/glaciers/data/SurfaceVelocityData_utils.jl

```diff
@@ -0,0 +1,265 @@
+    }
+
+Grid velocity data onto the glacier grid following the prescribed mapping.
+This function maps the 3 dimensional surface velocities to the glacier grid.
+The provided surface velocities can be a `Rasters.FileArray` which happens when the
+`RasterStack` is instantiated in lazy mode. In this situation, only the smallest
+cube that contains all the needed data to construct the mapping is read from disk.
+The returned velocity variables have shape `(nTimes, nx, ny)` where `nTimes` is the
+number of time steps and `(nx, ny)` is the size of the glacier grid.
+
+Arguments:
+- `glacier::G`: Glacier instance which determines the glacier on which the
+    velocities are projected onto.
+- `latitudes::Vector{F}`: Vector of latitude values of the original surface
+    velocity grid.
+- `longitudes::Vector{F}`: Vector of longitude values of the original surface
+    velocity grid.
+- `vx::Union{FileArray, Array{Union{Missing, F}, 3}}`: X component of the original
+    surface velocities. It can be either a `Rasters.FileArray` if the datacube is
+    read in lazy mode, or a plain 3 dimensional array.
+- `vy::Union{FileArray, Array{Union{Missing, F}, 3}}`: Y component of the original
+    surface velocities. It can be either a `Rasters.FileArray` if the datacube is
+    read in lazy mode, or a plain 3 dimensional array.
+- `mapping::VM`: Mapping to use.
+
+Returns:
+- `xG`: A vector that gives the x coordinates of the glacier grid.
+- `yG`: A vector that gives the y coordinates of the glacier grid.
+- `vxG`: A 3 dimensional array of the x component of the velocity gridded onto the
+    glacier grid.
+- `vyG`: A 3 dimensional array of the y component of the velocity gridded onto the
+    glacier grid.
+"""
+function grid(
+    glacier::G,
+    latitudes::Vector{F},
+    longitudes::Vector{F},
+    vx::Union{FileArray, Array{Union{Missing, F}, 3}},
+    vy::Union{FileArray, Array{Union{Missing, F}, 3}},
+    mapping::VM
+) where {
+    G <: AbstractGlacier,
+    F <: AbstractFloat,
+    VM <: VelocityMapping,
+    FileArray <: Rasters.FileArray
+}
+    params_projection = glacier.params_projection
+    transformReverse(lat,lon) = ReverseUTMercator(
+        lat, lon;
+        k=params_projection["k"],
+        cenlon=params_projection["lon_0"], cenlat=params_projection["lat_0"],
+        x0=params_projection["x_0"], y0=params_projection["y_0"]
+    )
+    xG = map(x -> x.x.val, transformReverse.(glacier.cenlat, glacier.Coords["lon"]))
+    yG = map(x -> x.y.val, transformReverse.(glacier.Coords["lat"], glacier.cenlon))
+    cenlatVel = (latitudes[end]+latitudes[begin])/2
+    cenlonVel = (longitudes[end]+longitudes[begin])/2
+    xV = map(x -> x.x.val, transformReverse.(cenlatVel, longitudes))
+    yV = map(x -> x.y.val, transformReverse.(latitudes, cenlonVel))
+
+    velType = eltype(vx)
+    vxG = zeros(velType, glacier.nx, glacier.ny, size(vx,3))
+    vyG = zeros(velType, glacier.nx, glacier.ny, size(vx,3))
+
+    if mapping.spatialInterp=="nearest"
```

Comment:
Small suggestion: Use a symbol `:nearest` rather than a string for literal variables.